### DISCUSSION TITLE
control loop element of valves

### DISCUSSION THREAD

**Discussion by DerMo92** [nliaudat/floor-heating-controller#8] [Q&A] [answered]
Dear @nliaudat I was looking at the projekt and plan to integrade in my HA-Server. 
Situation:
I have NS Panels with the Blackymas Blueprint and want to integrate the Temp-sensor in your solution and your solution in the Blackymas Blueprint. 
Question: 
1. I wander what kind of controler is used. I can't finde any P, I or PI - controler calculation. Can you help me here?
2. I also wander why there are only 7 temperatur sensors in the sensor_temperature.yaml Since I understand that you simply put the entity of the temperatur sensor in the yaml and it is then connected to each of a valve output. 

Can you help me out here?
Thank You 
BR Moritz

**Comment by nliaudat** [comment, marked as answer]
Hi, 

I mean you'll find all informations on how to control the board with hassio in https://github.com/nliaudat/esp32_8ch_motor_shield/issues/4

The main component used is esphome climate which you can setup in https://github.com/nliaudat/floor-heating-controller/blob/main/config.yaml. **The simple way is to update the climate temperature you want**. The controller will adjust valve position to reach that temperature.

Each climate component has it's own temperature sensor. In the sensor_temperature.yaml, you'll find 7 samples I use, but also external  sensors like DHT, 1-wire, etc. 

I've tested PID solutions, but it's not the good idea as : 
1 : the PID climate make the valve activate too much
2 : Very difficult to find good parameters

**Reply by DerMo92** [reply]
Hi @nliaudat, 

thank you for your reply. 
I'll try with the both I think but first without the PDI and your calculation. 
Sorry asking again but I don't get a few things: 
So I ordered the PCB and flashed a general ESP home to the dev board. What I don't get Is:
1. Config. in ESPNAME.Yaml to be changed acc. to which file?
2. How is the Structure where do I place the extra files? So Within wich folder do you ref. to (e.g. do You put these in CustomComponents/Floorheatin/config.yaml, sensor.temperature.yaml etc?)
3. Which is what? So in the Config.yaml do you have the Temperature Sensor I refer To and in the Temperatures.Yaml the ones that are been created to be used as Climate control?

Sorry again 
thank you in advance
*If you want I will provided you some Forage and help to extend the docu.
BR

**Comment by nliaudat** [comment]
Hi, 

The simple way is to flash the firmware is : 
1) download and unzip that repo files in your computer
2) make changes to config.yaml and sensor_temperature.yaml following your needs
3) flash the firmware (you need to be connected with usb the 1st time)

```
cd \esphome_folder_you_download_from_that_repo
esphome run config.yaml
```

No need to link anything as in question 1 and 2. 

For question 3):
in config.yaml you define each sensor is used for each valve channel : 
`  ch1_thermostat_sensor: **corridor_temperature_sensor**`
The name corridor_temperature_sensor must match one of the temperature sensor defined in sensor_temperature.yaml

```
  - platform: homeassistant
    name: "Corridor temperature sensor from hassio"
    id: **corridor_temperature_sensor**
```
**corridor_temperature_sensor** must match in config.yaml and sensor_temperature.yaml

You have nothing to change in other files cause the substitutions in config.yaml are used in the other files like climate.yaml

**Reply by DerMo92** [reply]
Hi,
Thank you for the quick reply.
So I created it as an ESPHome device so far and changed the configuration in there to your config. File you use in the part in edit as
EspHome
![EspHome](https://github.com/nliaudat/floor-heating-controller/assets/98593122/497a5a0d-3892-4728-9615-5807d7394e11)

following:
`
packages: wifi: !include /config/custom_components/floorheating/wifi.yaml board: !include /config/custom_components/floorheating/board.yaml time: !include /config/custom_components/floorheating/time.yaml climate: !include /config/custom_components/floorheating/climate.yaml script_termostat_check: !include /config/custom_components/floorheating/script_termostat_check.yaml script_calibrate_valves: !include /config/custom_components/floorheating/script_calibrate_valves.yaml switch_gpio: !include /config/custom_components/floorheating/switch_gpio.yaml switch_others: !include /config/custom_components/floorheating/switch_others.yaml inputs: !include /config/custom_components/floorheating/inputs.yaml covers: !include /config/custom_components/floorheating/covers.yaml sensor_bemf: !include /config/custom_components/floorheating/sensor_bemf.yaml #back electromotive force (back EMF) sensor_adc: !include /config/custom_components/floorheating/sensor_adc.yaml sensor_temperature: !include /config/custom_components/floorheating/sensor_temperature_eg.yaml sensor_others: !include /config/custom_components/floorheating/sensor_others.yaml `
That seams to work, so question acc. where to put is not relevant anymore. I think If this works as well to your explanation of manual push.

The question I still have to your explanation.- Sorry
So I understand the one I use in Config.yaml need to mach one in the sensor_temperature.yaml - ok but does this also need to mach the name of the already existing sensor or needs to be defined different?
Also there is a difference between entity_Id and the id it selfe. There are in the example even differneces in these. 

`- platform: homeassistant
name: "Corridor temperature sensor from hassio"
entity_id: sensor.corridor_temperature
id: corridor_temperature_sensor
entity_id: sensor.chambre_temperature
id: bedroom_temperature_sensor`

Since I don't get the difference beteween actual-temperature and target-temperature.

Ok I checked again the explanation:

Is it true as I undersand That you create all of them
` # float current_temp = id(${ch1_thermostat_name}_thermostat).current_temperature; # float target_temp_low = id(${ch1_thermostat_name}_thermostat).target_temperature_low; # float target_temp_high = id(${ch1_thermostat_name}_thermostat).target_temperature_high ;`

and I need to mach my existing real sensor mesured temperature with the one corridor_temperature_sensor.current_temperature ??
thank you in advance.

BR

**Reply by nliaudat** [reply]
You cannot install with web interface ....
Please follow https://esphome.io/guides/installing_esphome
and then compile.

**Reply by DerMo92** [reply]
Ok I'll try thank you for that hint and sorry for inconvenience. 

But still I have that question I described above - have you seen that? How is that with climate? 
Thank you in advance

The question I still have to your explanation.- Sorry
So I understand the one I use in Config.yaml need to mach one in the sensor_temperature.yaml - ok but does this also need to mach the name of the already existing sensor or needs to be defined different?
Also there is a difference between entity_Id and the id it selfe. There are in the example even differneces in these.

- platform: homeassistant name: "Corridor temperature sensor from hassio" entity_id: sensor.corridor_temperature id: corridor_temperature_sensor entity_id: sensor.chambre_temperature id: bedroom_temperature_sensor

Since I don't get the difference beteween actual-temperature and target-temperature.

Ok I checked again the explanation:

Is it true as I undersand That you create all of them
# float current_temp = id(${ch1_thermostat_name}_thermostat).current_temperature; # float target_temp_low = id(${ch1_thermostat_name}_thermostat).target_temperature_low; # float target_temp_high = id(${ch1_thermostat_name}_thermostat).target_temperature_high ;

and I need to mach my existing real sensor mesured temperature with the one corridor_temperature_sensor.current_temperature ??
thank you in advance.

BR

**Reply by nliaudat** [reply]
Please take a look at https://esphome.io/components/sensor/homeassistant.html + https://esphome.io/guides/configuration-types.html#substitutions and you'll understand

**Reply by DerMo92** [reply]
Hey thank you for your answer,

I understand the way it is looking. Yes but I don't understand. If the entity Id need to Mach the Is of the temperature in the room and the ID a name within the Config. At least that would make sense but what about the ID then? 
Example:
`-platform: homeassistant
  name: "Büro from Hassio"
  entity_id: sensor.ns_buero_temperature 
  id: buero_temperature_sensor`

So this would mean that 
I take the Value from homeassistant API
Platform: homeassistant
Give it a name: "Büro from Hassio"
Indicating it comes from home assistant.

Then I enter the entety_ID in home assistant which contains the temperature: ns_Buero_temperature 

And Am I right to then give a ID for internal reference in Config.yaml? 
Like ID: Buero_Temperature_sensor? 
That than need to Mach the name I invented in the Config? 

That is the point I still don't get. 
Sorry 
Thank you
BR

**Reply by DerMo92** [reply]
Hey,
So I tried that out and seams to work since it is getting the correct current Temperature in the Climate entity coming back. 
Maybe my thoughts were just too complicated. 

So now I need to group two channels in one Climate. I think you started a discussion about that already.
Hope I can come back if I get issues. 
Thank you


INFO: I did manage to flash initially via HA and the Web interface. 
BR

**Reply by nliaudat** [reply]
The simple way to group channels is to have the same temperature sensor and target temp ....

The other way it to make changes in script_thermostat_check.yaml to call a 2nd movement : 

```
                    auto call = id(CH2_cover).make_call();
                    call.set_position(target_position);
                    call.perform();
```

But if you want to control everything in hassio, you the 1st way is the best.
